tests/cli/test_validate_json_telemetry_and_dry_run.py

```
@@ -0,0 +1,104 @@
+
+
+def test_json_includes_fixes_metrics_and_dry_run(tmp_path: Path) -> None:
+    yml = (
+        'version: "0.1"\n'
+        "steps:\n"
+        '  - name: A\n    agent: { id: "flujo.builtins.stringify" }\n    input: "hello"\n'
+        '  - name: B\n    agent: { id: "flujo.builtins.stringify" }\n    input: "{{ previous_step.output }} {{ previous_step | to_json }}"\n'
+    )
+    f = tmp_path / "p.yaml"
+    f.write_text(yml)
+    res = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)